packages/web/src/components/gcds-radios/gcds-radios.tsx

```diff
@@ -0,0 +1,529 @@
+      this.lang = newValue;
+    }
+  }
+
+  private onChange = e => {
+    this.gcdsChange.emit(e.target.value);
+    this.value = e.target.value;
+    this.internals.setFormValue(e.target.value, 'checked');
+
+    const changeEvt = new e.constructor(e.type, e);
+    this.el.dispatchEvent(changeEvt);
+  };
+
+  /*
+   * Validate required properties
+   */
+  private validateRequiredProps() {
+    this.validateLegend();
+    this.validateName();
+
+    if (
+      this.errors.includes('name') ||
+      this.errors.includes('legend') ||
+      this.errors.includes('options')
+    ) {
+      return false;
+    }
+
+    return true;
+  }
+
+  async componentWillLoad() {
+    // Define lang attribute
+    this.lang = assignLanguage(this.el);
+
+    this.validateOptions();
+    this.validateRequiredProps();
+    this.validateErrorMessage();
+    this.validateValidator();
+
+    // Assign required validator if needed
+    requiredValidator(this.el, 'radio');
+
+    if (this.validator) {
+      this._validator = getValidator(this.validator);
+    }
+
+    this.inheritedAttributes = inheritAttributes(this.el, this.shadowElement);
+    this.initialValue = this.value ? this.value : null;
+
+    const valid = this.validateRequiredProps();
+
+    if (!valid) {
+      logError('gcds-radios', this.errors);
+    }
+  }
+
+  async componentDidUpdate() {
+    // Validate props again if changed after render
+    const valid = this.validateRequiredProps();
+
+    if (!valid) {
+      logError('gcds-radios', this.errors);
+    }
+  }
+
+  render() {
+    const {
+      lang,
+      name,
+      legend,
+      value,
+      required,
+      hint,
+      errorMessage,
+      disabled,
+      hasError,
+      inheritedAttributes,
+    } = this;
+
+    const fieldsetAttrs = {
+      'tabindex': '-1',
+      'aria-labelledby': 'radios-legend',
+    };
+
+    if (hint) {
+      const hintID = this.hint ? `radios-hint ` : '';
+      fieldsetAttrs['aria-labelledby'] =
+        `${fieldsetAttrs['aria-labelledby']} ${hintID}`.trim();
+    }
+
+    if (this.validateRequiredProps()) {
+      return (
+        <Host onBlur={() => this.onBlurValidate()}>
+          <fieldset class="gcds-radios__fieldset" {...fieldsetAttrs}>
+            <legend id="radios-legend" class="gcds-radios__legend">
+              {legend}
+              {required ? (
+                <span class="legend__required">{i18n[lang].required}</span>
+              ) : null}
+            </legend>
+
+            {hint ? (
+              <gcds-hint id="radios-hint" hint-id="radios">
+                {hint}
+              </gcds-hint>
+            ) : null}
+
+            {errorMessage ? (
+              <div>
+                <gcds-error-message id="radios-error" messageId="radios">
+                  {errorMessage}
+                </gcds-error-message>
+              </div>
+            ) : null}
+
+            {this.optionObject &&
+              this.optionObject.map(radio => {
```

Comment:
Would you see any value in moving the code for each radio into a subcomponent (like in breadcrumbs) to reduce the length of this file?

Reply:
I don't think it would be worth it to create a new component like `gcds-breadcrumbs-item` but I could try to make a [functional component](https://stenciljs.com/docs/functional-components) for the radio buttons.

Reply:
That works too :) Mainly thought it would be nice to break up the code a bit more to make it easier to read as one.